### ISSUE TITLE
Support to write to STAC APIs

### ISSUE THREAD

**Issue by alvaro-ps** [stac-utils/pystac-client#241]
Hi, thanks for the great work so far 👍🏻 💯 I was wondering if there is any plan to add functionality to write data to STAC APIs as well as reading from them?

**Comment by gadomski** [comment]
@philvarner would be an appropriate person to comment here, as he knows way more about the transactions API than I do.

**Comment by philvarner** [comment]
No plan that I'm aware of (and @gadomski would likely be more aware of a plan than I would).

**Comment by duckontheweb** [comment]
I'm working on a data catalog infrastructure right now that would benefit from this functionality, but I'm wondering what the best home would be for it.

@philvarner correct me if I'm wrong, but my interpretation of the [Transaction Fragment](https://github.com/radiantearth/stac-api-spec/tree/master/ogcapi-features/extensions/transaction#stac-api---transaction) in the API spec is that STAC API Transactions intends to align with OGC API Features transactions once it is out of the draft state. If that is the case (and we are not adding any additional functionality in the STAC API spec), then would it make more sense to support this in  [OWSLib](https://github.com/geopython/OWSLib), since that is the primary OGC API client? @tomkralidis what are your thoughts on this?

**Comment by tomkralidis** [comment]
@duckontheweb if that's the case then I think an addition in OWSLib makes sense and is most welcome!  Let me know once you start drafting things, as I can see Transactions in OGC API being applied to OGC API - Records (and potentially other OGC APIs) as well, so this can be implemented as a shared capability in the codebase.

**Comment by duckontheweb** [comment]
> Let me know once you start drafting things, as I can see Transactions in OGC API being applied to OGC API - Records (and potentially other OGC APIs) as well, so this can be implemented as a shared capability in the codebase.

Yeah, it would be nice to come up with some abstraction that works for multiple OGC API specs. It looks like there are [already transactions](https://geopython.github.io/OWSLib/usage.html#csw) built into the CSW module of OWSLib, but I'm not familiar enough with that codebase to know if it would translate to OGC API - Features and others.

**Comment by tomkralidis** [comment]
The CSW transaction code is based on the OGC CSW - Manager class.  In today's OGC API ecosystem, transactions are defined in [OGC API - Features - Part 4: Create, Replace, Update and Delete](http://docs.ogc.org/DRAFTS/20-002.html), hence this would make sense in `owslib.ogcapi.features`.  Perhaps we can discuss this in a dedicated OWSLib issue?

**Comment by duckontheweb** [comment]
After looking at the OWSLib code in more detail I think there would be value in implementing transactions separately in pystac-client. The usage pattern of OWSLib is very different from this library and I think it might be confusing for pystac-client users to have to install and learn another dependency in order to use the Transactions endpoints. 
I can work on implementing that in pystac-client and see if any of it is transferable to OWSLib.
